src/ethPriceLoop/db.eth_price.ts

```diff
@@ -0,0 +1,28 @@
+import { ETH_PRICE_TABLE } from '../constants';
+import { getDataSource } from '../db';
+import { EthPrice } from '../entities/IEthPrice';
+
+export async function getEthPriceCount() {
+  return await getDataSource().getRepository(EthPrice).count();
+}
+
+export async function persistEthPrices(prices: EthPrice[]) {
+  await getDataSource()
+    .getRepository(EthPrice)
+    .upsert(prices, ['timestamp_ms']);
+}
+
+export async function getClosestEthUsdPrice(date: Date): Promise<number> {
+  const timestampMs = date.getTime();
+  const price = await getDataSource().manager.query(
+    `
+      SELECT * from ${ETH_PRICE_TABLE}
+      WHERE timestamp_ms <= ?
+      ORDER BY timestamp_ms DESC
+      LIMIT 1
+  `,
+    [timestampMs]
+  );
+
+  return price[0].usd_price ?? 0;
```

Comment:
This would result to an error if there are 0 rows. `price[0]?.usd_price ?? 0` would be more correct.

Reply:
fixed